cetmix_tower_server/models/cx_tower_plan_line.py

```diff
@@ -187,3 +192,10 @@ def _skip(self, server, plan_log_record, **kwargs):
             is_skipped=True,
             **log_vals,
         )
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            current_datetime = datetime.now().strftime("%Y-%m-%d %H:%M:%S.%f")
+            vals["name"] = f"Plan line {current_datetime}"
```

Comment:
Why do we need to create a name for a plan line? It's not used anywhere

Reply:
You are right, but the generation method depends on this field. To avoid possible problems, we are forced to add the value this way. Also, the reference must be unique, so we include the date in them.

Reply:
I agree, however it's [related field](https://github.com/cetmix/cetmix-tower/blob/2711a9e613a0c3f7728cf176665e78f74ec8372a/cetmix_tower_server/models/cx_tower_plan_line.py#L22) 
Meanwhile [reference](https://github.com/cetmix/cetmix-tower/blob/2711a9e613a0c3f7728cf176665e78f74ec8372a/cetmix_tower_server/models/cx_tower_reference_mixin.py#L138) field is also used during the creation of the record. 
So I would suggest something like this:

```python
@api.model_create_multi
def create(self, vals_list):
     PlanModel = self.env["cx.tower.plan"]

     # Store the last line index per plan
     plan_line_index_dict = {}

     # Populate vals with references
     for vals in vals_list:
        plan_id = vals.get("plan_id")
        # Get stored line index for this plan and increment it
        line_index = plan_line_index_dict.get(plan_id, 1) + 1
        # Line reference = <plan_reference>_<line_index>
        vals.update({"reference": f"{PlanModel.browse(plan_id).reference}_{line_index}"})
        # Save updated line index
        plan_line_index_dict.update({plan_id: line_index})

    # Let main function do the job
    return super().create(vals_list)
```

@GabbasovDinar your thoughts on that?

Reply:
```python

from collections import defaultdict


@api.model_create_multi
def create(self, vals_list):
    PlanModel = self.env["cx.tower.plan"]

    # Extract all unique plan_ids from vals_list
    plan_ids = {vals.get("plan_id") for vals in vals_list if vals.get("plan_id")}

    # Fetch all plan references in a single query
    plans = PlanModel.browse(plan_ids)
    plan_refs = {plan.id: plan.reference for plan in plans}

    # Initialize line indices for each plan_id
    plan_line_index_dict = defaultdict(int)

    # Populate vals with references
    for vals in vals_list:
        plan_id = vals.get("plan_id")
        plan_line_index_dict[plan_id] += 1
        line_index = plan_line_index_dict[plan_id]
        vals["reference"] = f"{plan_refs[plan_id]}_{line_index}"

    # Delegate creation to the superclass method
    return super().create(vals_list)
```

Reply:
what about if plan_id not in vals ?

Reply:
and what about this [line](https://github.com/cetmix/cetmix-tower/blob/2711a9e613a0c3f7728cf176665e78f74ec8372a/cetmix_tower_server/models/cx_tower_reference_mixin.py#L135)?

Reply:
> what about if plan_id not in vals ? eg: [here](https://github.com/cetmix/cetmix-tower/blob/2711a9e613a0c3f7728cf176665e78f74ec8372a/cetmix_tower_server/tests/test_plan.py#L37)

@dessanhemrayev thank you for pointing on this. In this case we can create some default placeholder reference. Eg "line_with_no_plan"

Reply:
> and what about this [line](https://github.com/cetmix/cetmix-tower/blob/2711a9e613a0c3f7728cf176665e78f74ec8372a/cetmix_tower_server/models/cx_tower_reference_mixin.py#L135)?

@dessanhemrayev good point! I think we need to add a condition here. Eg `if "name" in vals: ...`